crates/holochain/src/conductor/handle.rs

```diff
@@ -389,16 +409,20 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
                 respond.respond(Ok(async move { Ok(signature) }.boxed().into()));
             }
             _ => {
-                let cell = self.cell_by_id(cell_id).await?;
-                cell.handle_holochain_p2p_event(event).await?;
+                self.with_cell(cell_id, |cell| {
+                    async move { Ok(cell.handle_holochain_p2p_event(event).await?) }.boxed()
+                })
+                .await?
             }
         }
         Ok(())
     }
 
     async fn call_zome(&self, call: ZomeCall) -> ConductorApiResult<ZomeCallResult> {
-        let cell = self.cell_by_id(&call.cell_id).await?;
-        Ok(cell.call_zome(call, None).await?)
+        self.with_cell(&call.cell_id.clone(), |cell| {
+            async move { Ok(cell.call_zome(call, None).await?) }.boxed()
+        })
+        .await
```

Comment:
Now your holding a read lock on the conductor for the entire zome call. This is why we put the Cell into an Arc so we could clone it without holding the read lock.